components/ArticleBoard/ArticleBoard.tsx

```diff
@@ -0,0 +1,57 @@
+import React, { useState, useEffect } from 'react';
+import ArticleList from '@/components/ArticleList/ArticleList';
+import { ArticleData } from '@/types/types';
+
+interface ArticleBoardProps {
+  articles: ArticleData[];
+}
+
+const ArticleBoard = ({ articles }: ArticleBoardProps) => {
+  const [sortedArticles, setSortedArticles] = useState<ArticleData[]>([]);
+  const [sortOrder, setSortOrder] = useState('newest');
+  const [searchTerm, setSearchTerm] = useState('');
+
+  useEffect(() => {
+    filterAndSortArticles();
+  }, [articles, sortOrder, searchTerm]);
+
+  const filterAndSortArticles = () => {
+    let filtered = articles.filter(article => 
+      article.title.toLowerCase().includes(searchTerm.toLowerCase())
+    );
+
+    if (sortOrder === 'likes') {
+      filtered.sort((a, b) => b.likeCount - a.likeCount);
+    } else {
+      filtered.sort((a, b) => new Date(b.createdAt).getTime() - new Date(a.createdAt).getTime());
+    }
+    setSortedArticles(filtered);
```

Comment:
```suggestion
    const filtered = articles.filter(article => 
      article.title.toLowerCase().includes(searchTerm.toLowerCase())
    );

    if (sortOrder === 'likes') {
     const filteredByLikes=  filtered.sort((a, b) => b.likeCount - a.likeCount);
       setSortedArticles(filteredByLikes);
       return
    } else {
     const filteredByOther =  filtered.sort((a, b) => new Date(b.createdAt).getTime() - new Date(a.createdAt).getTime());
      setSortedArticles(filteredByLikes);
       return
    }
```
위처럼 코드를 early Return 하는방법으로 코드를 쓰면 가독성이 올라가요.
왜냐하면 은우님이 작성하신 방법도 좋은 방법이지만 해당 함수를 끝까지 읽어야만 해당 동작을 확인할 수 있는데 얼리리턴하면 필요한 부분만 읽어도 돼요!